packages/react-date-time/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,253 @@
+  beforeMinDateErrorMessage?: string;
+  /** Error message to display when the date is provided after the provided maxDate. */
+  afterEndDateErrorMessage?: string;
+}
+
+interface DatePickerState {
+  invalid: boolean;
+  invalidText: string;
+}
+
+function toNumber(str: string) {
+  if (str) {
+    return Number(str);
+  }
+  return NaN;
+}
+// Supports (yyyy|mm|dd) to parse string to Date object
+function parseYYYYMMDD(date: string, format: string) {
+  let year = NaN;
+  let month = NaN;
+  let day = 1;
+  let index = 0;
+  if (date.length === format.length) {
+    for (const formatBlock of format.split(/(yyyy|mm|dd)/i)) {
+      if (formatBlock.toLowerCase() === 'yyyy') {
+        const yearString = date.substr(index, 4);
+        if (yearString.length === 4) {
+          year = toNumber(yearString);
+        }
+      } else if (formatBlock.toLowerCase() === 'mm') {
+        month = toNumber(date.substr(index, 2)) - 1;
+      } else if (formatBlock.toLowerCase() === 'dd') {
+        day = toNumber(date.substr(index, 2));
+      }
+      index += formatBlock.length;
+    }
+  }
+
+  // Year and month are required
+  return new Date(year, month, day);
+}
+
+// Supports (yyyy|mm|dd) to format Date object to string
+function formatYYYYMMDD(date: Date, format: string) {
+  let res = '';
+  for (const formatBlock of format.split(/(yy?y?y?|mm|dd?)/i)) {
+    if (['y', 'yy', 'yyy', 'yyyy'].includes(formatBlock.toLowerCase())) {
+      res += String(date.getFullYear()).padStart(4, '0');
+    } else if (['mm'].includes(formatBlock.toLowerCase())) {
+      res += String(date.getMonth() + 1).padStart(2, '0');
+    } else if (['d', 'dd'].includes(formatBlock.toLowerCase())) {
+      res += String(date.getDate()).padStart(2, '0');
+    } else {
+      res += formatBlock;
+    }
+  }
+  return res;
+}
+
+function isValid(date: Date) {
+  return !isNaN(date.getDate());
+}
+
+export class DatePicker extends React.Component<DatePickerProps, DatePickerState> {
+  static displayName = 'DatePicker';
+  static defaultProps = {
+    className: '',
+    invalid: false,
+    minDate: '',
+    maxDate: '',
+    dateFormat: 'MM/DD/YYYY',
+    isDisabled: false,
+    locale: DatePickerLocales.en,
+    placeholder: 'mm/dd/yyyy',
+    value: '',
+    'aria-label': 'Date picker'
+  };
+
+  constructor(props: DatePickerProps) {
+    super(props);
+    this.state = {
+      invalid: this.props.invalid,
+      invalidText: ''
+    };
+  }
+
+  private ref = React.createRef<HTMLDivElement>();
+  private inputEl = React.createRef<HTMLInputElement>();
+  private calendar: any = null;
+
+  componentDidMount() {
+    const { dateFormat, minDate, maxDate, locale, value } = this.props;
+
+    this.calendar = flatpickr(this.inputEl.current, {
+      dateFormat,
+      locale: (FlatpickrLanguages as any)[locale],
+      minDate,
+      maxDate,
+      defaultDate: value,
+      ariaDateFormat: dateFormat,
+      allowInput: true,
+      static: true,
```

Comment:
not sure you'd want this to be true, if I understand this option correctly it would place the datepicker inline as opposed to appending it to document.body. This would cause the picker not to break out of a Modal or Wizard for example

Reply:
hm... okay. I grouped it with the input based on core's request. I'll double check with them.

Reply:
Should we expose that as a prop like we do with `menuAppendTo` in Dropdown and Select?

Reply:
that'd take a bit more experimentation. possibly.

Reply:
only because I need to figure out how that could impact how flatpickr handles positioning.

Reply:
okay. it looks like flatpickr has incorporated a feature that will give us more fine tuned control over the positioning of the calendar relative to an element. Right now, though it gives the calendar absolute position and calculates its `top` and `left` position based on the element the consumer tells it to append to. This causes problems for us based on PatternFly's current page setup. so once the latest feature gets released, we can probably append to body or another container, but for now, this is the best way to position the calendar - inline with the input.